src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -8583,6 +8583,554 @@ static void Main()
+                Diagnostic(ErrorCode.ERR_NoEntryPoint).WithLocation(1, 1)
+                );
+        }
+
+        [Fact]
+        public void MethodModifiers_31()
+        {
+            var source1 =
+@"
+public partial interface I1
+{
+    static partial void M1<T>() where T : struct;
+    static partial void M1<T>() {}
+
+    static partial void M2(params int[] x);
+    static partial void M2(int[] x) {}
+
+    static partial void M3();
+    partial void M3() {}
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.NetStandardLatest);
+
+            compilation1.VerifyDiagnostics(
+                // (5,25): error CS0761: Partial method declarations of 'I1.M1<T>()' have inconsistent type parameter constraints
+                //     static partial void M1<T>() {}
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("I1.M1<T>()").WithLocation(5, 25),
+                // (8,25): error CS0758: Both partial method declarations must use a params parameter or neither may use a params parameter
+                //     static partial void M2(int[] x) {}
+                Diagnostic(ErrorCode.ERR_PartialMethodParamsDifference, "M2").WithLocation(8, 25),
+                // (11,18): error CS0763: Both partial method declarations must be static or neither may be static
+                //     partial void M3() {}
+                Diagnostic(ErrorCode.ERR_PartialMethodStaticDifference, "M3").WithLocation(11, 18)
+                );
+        }
+
+        [Fact]
+        public void MethodModifiers_32()
+        {
+            var source1 =
+@"
+partial interface I1
+{
+    partial void M1();
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.DesktopLatestExtended);
+
+            compilation1.VerifyDiagnostics();
+
+            var source2 =
+@"
+partial interface I1
+{
+    partial void M1() {}
+}
+";
+            var compilation2 = CreateCompilation(source1 + source2, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.DesktopLatestExtended);
+
+            compilation2.VerifyDiagnostics(
+                // (9,18): error CS8701: Target runtime doesn't support default interface implementation.
+                //     partial void M1() {}
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, "M1").WithLocation(9, 18)
+                );
+        }
```

Comment:
Consider tests when an explicit access modifier is used on a `partial` method.

Reply:
Consider a test where a non-void return is used.

---
In reply to: [259462424](https://github.com/dotnet/roslyn/pull/33594#discussion_r259462424) [](ancestors = 259462424)

Reply:
Both covered in MethodModifiers_26

---
In reply to: [259462522](https://github.com/dotnet/roslyn/pull/33594#discussion_r259462522) [](ancestors = 259462522,259462424)